src/System.Xml.XDocument/tests/xNodeBuilder/CommonTests.cs

```diff
@@ -3599,17 +3603,21 @@ public void comment_5()
                 public void WriteCommentWithDoubleHyphensInValue()
                 {
                     XDocument doc = new XDocument();
-                    XmlWriter w = doc.CreateWriter();
-                    w.WriteStartElement("Root");
-                    w.WriteComment("test --");
-                    w.WriteEndElement();
-                    w.Dispose();
+                    using (XmlWriter w = doc.CreateWriter())
+                    {
+                        w.WriteStartElement("Root");
+                        w.WriteComment("test --");
+                        w.WriteEndElement();
+                    }
+
                     try
                     {
-                        if (CompareReader(doc, "<Root><!--test - - --></Root>"))
+                        string expectedXml = "<Root><!--test - - --></Root>";
+
+                        using (XmlReader reader = doc.CreateReader())
                         {
-                            throw new TestException(TestResult.Failed, "");
-                        } //by design
+                            Assert.Equal(expectedXml, MoveToFirstElement(reader).ReadOuterXml());
+                        }
```

Comment:
It's not clear to me why this has been added. But currently, this line is not hit in any of these tests.